compiler/circle-execution-plan/src/ExecutionPlanner.cpp

```diff
@@ -358,34 +359,42 @@ void ExecutionPlanner::create_alloc_node_inform_vector(bool null_consts, bool nu
       _alloc_node_inform_vector[i].size = node_size;
     }
 
-    // Im2col
-    auto opcode = circle_node->opcode();
-    if (opcode == luci::CircleOpcode::CONV_2D)
+    // Scratchpad If needed
+    uint32_t scratchpad_size = 0;
+    switch (circle_node->opcode())
     {
-      auto conv = loco::must_cast<const luci::CircleConv2D *>(circle_node);
-      auto im2col_size = compute_im2col_size(conv);
-      if (im2col_size > 0)
+      case luci::CircleOpcode::CONV_2D:
       {
-        AllocationNodeInformation temp_alloc;
-
-        if (null_im2col)
-        {
-          temp_alloc.size = 0;
-        }
-        else
-        {
-          temp_alloc.size = im2col_size;
-        }
-
-        temp_alloc.first_node = i - 1;
-        temp_alloc.last_node = i + 1;
-        temp_alloc.node_num = i;
-        temp_alloc.is_temp = true;
-
-        _alloc_node_inform_vector.push_back(temp_alloc);
-        _alloc_node.push_back(i);
-        _dealloc_node.push_back(i);
+        auto conv = loco::must_cast<const luci::CircleConv2D *>(circle_node);
+        scratchpad_size = compute_im2col_size(
```

Comment:
Since `im2col` uses only with `CircleConv` I think better to handle `null_im2col` here:

```suggestion
        scratchpad_size = null_im2col ? 0 : compute_im2col_size(
```

And then remove redundant `if (null_im2col)` statement bellow.

Reply:
Done